src/Microsoft.SqlTools.ServiceLayer/Migration/MigrationService.cs

```diff
@@ -402,6 +405,50 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
+        internal async Task HandleSaveSkuRecommendationsRequest(
+            SaveSkuRecommendationsParams parameters,
+            RequestContext<SaveSkuRecommendationsResult> requestContext)
+        {
+            const string SkuRecommendationJsonFileExt = ".json";
+            const string SkuRecommendationHtmlFileExt = ".html";
+            string timeStamp = DateTime.UtcNow.ToString("yyyyMMddHHmmss");
+
+            string[] skuRecommendationFileNames = new string[6];
+
+            try
+            {
+                // Save results
+                SkuRecommendationReport reportMI = new SkuRecommendationReport(
+                    new Dictionary<SqlInstanceRequirements, List<SkuRecommendationResult>> { [parameters.InstanceRequirements] = parameters.SqlMiRecommendationResults },
+                    AzureSqlTargetPlatform.AzureSqlManagedInstance.ToString());
+                string recommendationMIFileName = String.Format("SkuRecommendation{0}-{1}.json", AzureSqlTargetPlatform.AzureSqlManagedInstance.ToString(), timeStamp);
+                ExportRecommendationResultsAction.ExportRecommendationResults(reportMI, SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, false, recommendationMIFileName);
+                skuRecommendationFileNames[0] = recommendationMIFileName + SkuRecommendationJsonFileExt;
+                skuRecommendationFileNames[1] = recommendationMIFileName + SkuRecommendationHtmlFileExt;
+
+                SkuRecommendationReport reportVM = new SkuRecommendationReport(
+                    new Dictionary<SqlInstanceRequirements, List<SkuRecommendationResult>> { [parameters.InstanceRequirements] = parameters.SqlVmRecommendationResults },
+                    AzureSqlTargetPlatform.AzureSqlManagedInstance.ToString());
+                string recommendationVMFileName = String.Format("SkuRecommendation{0}-{1}.json", AzureSqlTargetPlatform.AzureSqlVirtualMachine.ToString(), timeStamp);
+                ExportRecommendationResultsAction.ExportRecommendationResults(reportVM, SqlAssessmentConfiguration.ReportsAndLogsRootFolderPath, false, recommendationVMFileName);
+                skuRecommendationFileNames[0] = recommendationVMFileName + SkuRecommendationJsonFileExt;
+                skuRecommendationFileNames[1] = recommendationVMFileName + SkuRecommendationHtmlFileExt;
```

Comment:
I see you store the `skuRecommendationFileNames` in index 0 and 1 for the MI file names above, but then overwrite index 0 and 1 again for the VM file names here? Did you mean to use index 2 and 3 instead? Either way, see my other comment about using List instead of an array, maybe that would simplify things